src/openrct2/core/Compression.cpp

```diff
@@ -138,4 +139,137 @@ namespace OpenRCT2::Compression
         inflateEnd(&strm);
         return true;
     }
+
+    /*
+     * Modified copy of ZSTD_COMPRESSBOUND / ZSTD_compressBound() from zstd 1.5.7, with the argument
+     * type changed from size_t (which may be only 32 bits) to uint64_t, and removes the error handling.
+     */
+    static uint64_t zstdCompressBound(uint64_t length)
+    {
+        return length + (length >> 8)
+            + ((length < (128uLL << 10)) ? (((128uLL << 10) - length) >> 11) /* margin, from 64 to 0 */ : 0uLL);
+    }
+
+    bool zstdCompress(IStream& source, uint64_t sourceLength, IStream& dest, ZstdMetadata metadata, int16_t level)
+    {
+        Guard::Assert(sourceLength <= source.GetLength() - source.GetPosition());
+
+        size_t ret;
+        StreamReadBuffer sourceBuf(source, sourceLength, ZSTD_CStreamInSize());
+        StreamWriteBuffer destBuf(dest, zstdCompressBound(sourceLength), ZSTD_CStreamOutSize());
+        unsigned metaFlags = static_cast<unsigned>(metadata);
```

Comment:
uint32_t?

Reply:
I could switch this type, but I had used `unsigned` for a reason. Specifically, `metadata`'s type is the enum type `ZstdMetadata`, which uses the default underlying type of `int`. Since I am performing bitwise operations on the result, I prefer having an unsigned result (since it's less likely to have undefined behavior). As such, the type to cast to should be `std::make_unsigned_t<std::underlying_type_t<ZstdMetadata>>`, which is `unsigned`. If I *do* change this to `uint32_t` (for stylistic reasons I assume), I should also change the underlying type of `ZstdMetadata` to `[u]int32_t` to match.